src/main/scala/io/iohk/ethereum/domain/Account.scala

```diff
@@ -1,21 +1,26 @@
 package io.iohk.ethereum.domain
 
 import akka.util.ByteString
-import org.spongycastle.util.encoders.Hex
+
+object Account {
+  val Empty = Account(0, 0, ByteString.empty, ByteString.empty)
+}
 
 case class Account(
   nonce: BigInt,
   balance: BigInt,
   storageRoot: ByteString,
   codeHash: ByteString) {
 
-  override def toString: String = {
-    s"""Account{
-         |nonce: $nonce
-         |balance: $balance wei
-         |storageRoot: ${Hex.toHexString(storageRoot.toArray[Byte])}
-         |codeHash: ${Hex.toHexString(codeHash.toArray[Byte])}
-         |}
-       """.stripMargin
-  }
+  def updateBalance(value: BigInt): Account =
+    copy(balance = balance + value)
+
+  def increaseNonce: Account =
+    copy(nonce = nonce + 1)
+
+  def withCode(codeHash: ByteString): Account =
+    copy(codeHash = codeHash)
```

Comment:
That's not 100% correct. On contract creation, first an account with balance and no code gets created. Then the code that was passed in the message call gets executed in the new account's context. The return data of that execution becomes the new account's code.

After that there is no direct way to modify an account's code, but that's a direct consequence of the set of instructions that we have available, so I don't think there's anything to enforce here. Would you agree?

Food for thought: what happens when 
1. contract A calls contract B, and as a result B self-destructs
2. after that contract A creates a new account with the same address as B

Following Yellow Paper by the letter that would mean B is not recreated, but A pays the gas and value cost for its creation, and consequently some amount Ether vanishes. @whysoserious